sim/battle.ts

```diff
@@ -1219,7 +1219,9 @@ export class Battle {
 			for (let i = 0; i < this.sides.length; i++) {
 				const side = this.sides[i];
 				if (!side.pokemonLeft) continue;
-				const activeData = side.active.map(pokemon => pokemon?.getMoveRequestData());
+				let active = side.active;
+				if (this.gameType === 'rotation') active = active.concat(side.pokemon.slice(1, 3));
```

Comment:
Given the number of times that you use it, I think possibly there should be a property or method of `side` that returns three Pokémon when playing a rotation battle, so you can use that where you're currently having to special-case rotation battles. (Here you would just use it directly, but in other places you would use the length instead of special casing 3 or `this.active.length`.)